test/proxied.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+const NCMTestRunner = require('./lib/test-runner.js')
+
+// A polyfill for Node.js EE#once().
+const once = require('events.once')
+
+const { spawn } = require('child_process')
+const path = require('path')
+
+NCMTestRunner.test('api requests respect ENV proxy settings', async (runner, t) => {
+  const proc = spawn(
+    process.execPath,
+    [
+      path.join(__dirname, 'lib', 'http-proxy-bin.js'),
+      `http://localhost:${runner.port}`,
+      0
+    ],
+    {
+      stdio: ['pipe', 'pipe', 'pipe', 'ipc']
+    }
+  )
+
+  if (process.env.NCM_DEV === true) {
+    proc.stdout.pipe(process.stdout)
+    proc.stderr.pipe(process.stderr)
+  }
+
+  proc.on('error', err => {
+    proc.kill()
+    t.fail('Proxy server emitted an error!', err)
+  })
+
+  const [ procPort ] = await once(proc, 'message')
+
+  try {
+    await runner.execP('details npm @ 6.8.0', {
+      http_proxy: `http://localhost:${procPort}`
+    })
+  } catch (err) {
+    const { stdout, stderr } = err
```

Comment:
Since this command results in a non-zero exit code, this is how it needs to be handled with `await`.